### PR TITLE
fix: route padded and renamed JAX JSON checkpoints

### PR DESCRIPTION
**Pull request by mldangelo-oai**
## Summary

- recognize structured JAX/Orbax JSON checkpoint metadata under misleading suffixes and route it through `jax_checkpoint`
- stream top-level identity discovery for oversized renamed JSON with bounded memory, so padding before `framework: jax` cannot suppress routing
- fail closed for identified JSON beyond the bounded full-analysis budget rather than calling `json.load` on an unbounded document
- preserve `ajax` and arbitrary JSON near-match exclusions across direct scans, skip filtering, and directory scans

## Why

A malicious JSON checkpoint is analyzed when named `bad.checkpoint`, but identical content renamed to `bad.jpg` could previously be classified as `unknown` and skipped. During audit of this draft, an oversized renamed checkpoint also stayed clean if more than 1 MiB of padding appeared before its JAX identity field:

```text
late-padded bad.jpg before repair: unknown -> exit 0
late-padded bad.jpg after repair:  jax_checkpoint -> inconclusive -> exit 2
late-padded ajax.jpg after repair: unknown -> exit 0
```

Small identified JSON still reaches existing JSON security analysis and reports concrete suspicious-pattern findings. For oversized identified JSON, the scanner cannot safely prove findings without an unbounded parse, so it reports incomplete coverage, is not cached as a clean result, and returns operational exit `2`.

## Implementation

- `modelaudit/utils/file/detection.py` retains strict parsing for JSON within the 1 MiB structural-analysis budget and uses bounded-memory streaming only to identify top-level JAX/Orbax identity fields in oversized renamed documents.
- `modelaudit/scanners/jax_checkpoint_scanner.py` marks oversized identified JSON as `inconclusive` with reason `jax_json_checkpoint_analysis_size_limit`.
- Tests cover small malicious routing, oversized late-identity routing, oversized `ajax` negatives, Orbax marker handling, skip filtering, directory scanning, exit-code behavior, and cache non-persistence.

## Related Audit Work

- #1280 handles the adjacent renamed Flax MessagePack content-routing miss independently.

## Validation

- `ruff format modelaudit/ packages/modelaudit-picklescan/src packages/modelaudit-picklescan/tests tests/`: clean
- `ruff check --fix modelaudit/ packages/modelaudit-picklescan/src packages/modelaudit-picklescan/tests tests/`: clean
- `mypy modelaudit/ packages/modelaudit-picklescan/src packages/modelaudit-picklescan/tests tests/`: clean (`451` files)
- `npx prettier --check CHANGELOG.md README.md docs/user/compatibility-matrix.md`: clean
- `PROMPTFOO_DISABLE_TELEMETRY=1 pytest tests/scanners/test_jax_checkpoint_scanner.py tests/utils/file/test_filetype.py tests/utils/file/test_file_filter.py tests/test_core.py tests/test_directory_file_filtering.py -q`: `342 passed`
- `PROMPTFOO_DISABLE_TELEMETRY=1 pytest -n auto -m "not slow and not integration" --maxfail=1`: `5706 passed, 16 skipped`

### CODE REVIEW

**Comment by mldangelo-oai** [comment]
Audit follow-up: the draft originally still returned a clean result for an oversized renamed JSON checkpoint when more than 1 MiB of padding preceded its top-level JAX identity field. On the repaired head `34e0d51b`, that late-padded sample routes to `jax_checkpoint` and returns explicit incomplete coverage with exit `2`; an equivalently padded `ajax` near match remains `unknown` with exit `0`.

The correction uses bounded-memory streaming for oversized identity discovery and avoids an unbounded JSON parse by failing closed once the full-analysis budget is exceeded. Regressions cover direct scan, cache non-persistence, skip filtering, directory scanning, and Orbax marker preservation.

Local validation: full Ruff and Prettier checks clean; full mypy clean (`451` files); required pytest lane `5706 passed, 16 skipped`.

**Comment by mldangelo-oai** [comment]
@codex review

**Comment by mldangelo-oai** [comment]
Follow-up after merge:

- The bounded/unbounded routing findings, long identity bypass, and JSON `.ckpt` / `.pickle` overlap findings are addressed in the merged #1281 implementation.
- The `ScanResult.success` suggestion is intentionally not applied: ModelAudit treats a completed scan with a security finding as successful execution with exit code `1`; `success=False` and exit code `2` are reserved for incomplete or operational scan coverage.
- The post-merge oversized JSON pattern-loss finding is addressed by #1380, which reports visible bounded-prefix threats while remaining fail-closed for the unread tail. During review it was additionally hardened for reread failures and caching, nested documentation context, arrays, truncated and JSON-escaped strings, malformed trailing roots, and complete-root duplicate-key semantics.
- The suggestion to skip an oversized prefix after seeing a visible non-JAX identity is intentionally not applied. Within the read budget, that prefix can be byte-identical to a file with a later hidden JAX identity; skipping it would reopen a padded content-routing bypass. #1380 adds regression coverage for this fail-closed decision.

Validation for #1380 includes the full required lane (`6364 passed, 16 skipped, 22 warnings`), affected JAX/routing/core coverage (`687 passed`), live CLI proofs for encoded/truncated threats and negative near-matches, and a final independent review with no remaining actionable finding.
